osl/preprocessing/batch.py

```diff
@@ -389,6 +398,8 @@ def run_proc_chain(infile, config, outname=None, outdir=None, ret_dataset=True,
                     traceback.print_tb(ex_traceback, file=f)
             return 0
 
+    dataset = append_preprocinfo(dataset, config)
```

Comment:
I'd move this to the end of `run_proc_chain` so the config is storedd even if we're just running in an interactive session without saving to disk

Reply:
That's where it is now, no? I'm calling `append_preprocinfo` right before potentially writing to disk. So the data saved to disk should contain the preprocinfo, and if the dataset is not saved but returned to the user, it should also contain it.

Reply:
you're right! I was fooled by githubs helpful code-collapsing....